app/views/guests/show.html.erb

```diff
@@ -0,0 +1,17 @@
+All Guests
+
+name: <%= @guest.name %>
+
+<p> Rooms </p>
+<ul>
+  <% @rooms.each do |room| %>
+    <li>suite: <%= room.suite%>, rate: <%= room.rate %> Hotel Name: <%= room.hotel.name%>
+  <% end %>
+</ul>
+
+Add Room:
+<% form_with url: "guests/#{@guest.id}", method: :post, local: true do |f| %>
+  <%= f.label "ID: "%>
+  <%= f.number_field :room_id %>
+  <%= f.submit "Submit"%>
+  <% end %>
```

Comment:
We mustn't forget about the HTML. Always put non-rendered ERB in an HTML tag.

```ruby
<h1>All Guests</h1>

<p>name: <%= @guest.name %></p>

<p>Add Room:</p>
```

This isn't just a plaintext document, this is an HTML file and therefore requires that all tags be surrounded by some HTML element. The HTML element that you plan to use is up to you as long as it is semantically correct.

---

The route in your form is not a RESTful route. Currently, the way that the route is formatted it looks something like this:

```ruby
post "/guests/:id"
```

This does not follow RESTful convention as the path for a POST never ends in an ID. The current route have is for `show`, `update`, and `delete` actions.

We need to remember what action we are trying to perform using the form. We want to create an association between a room and a guest. So essentially create a new entry in the RoomGuests join table. Your route should look something like this:

```ruby
<% form_with url: "guests/#{@guest.id}/rooms", method: :post, local: true do |f| %>
```

This route would resolve to something like this: `post "guests/:id/rooms"`

---

Last thing, the `f.label` should also have a name attribute that is the same as the field that the label is being used for.

```ruby
<%= f.label :room_id, "ID: " %>
```